diff --git a/CMakeLists.txt b/CMakeLists.txt
index d740938..34ada08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,7 @@ juce_add_plugin(SimpleReverb
     PLUGIN_CODE "Srvb"
     FORMATS "VST3" "AU" "Standalone" 
     VST3_CATEGORIES "Fx" 
+    NEEDS_CURL "TRUE"
     AU_MAIN_TYPE "kAudioUnitType_Effect")
 
 target_compile_features(SimpleReverb PUBLIC cxx_std_17)
@@ -22,7 +23,59 @@ target_compile_definitions(SimpleReverb PUBLIC
 add_subdirectory(Resources)
 add_subdirectory(Source)
 
-target_link_libraries(SimpleReverb PUBLIC 
+# begin AAP specifics -->
+set (APP_NAME SimpleReverb)
+set (OBOE_DIR External/JUCE/modules/juce_audio_devices/native/oboe)
+
+# They are needed in desktop too, for generate-aap-metadata.
+include_directories(
+        "${AAP_DIR}/include"
+)
+
+message("AAP_DIR: ${AAP_DIR}")
+message("AAP_JUCE_DIR: ${AAP_JUCE_DIR}")
+juce_add_modules(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+
+if (ANDROID)
+        add_library("cpufeatures" STATIC "${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c")
+        set_source_files_properties("${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c" PROPERTIES COMPILE_FLAGS "-Wno-sign-conversion -Wno-gnu-statement-expression")
+
+        # dependencies
+        find_library(log "log")
+        find_library(android "android")
+        find_library(glesv2 "GLESv2")
+        find_library(egl "EGL")
+        set(cpufeatures_lib "cpufeatures")
+
+        target_include_directories(${APP_NAME} PRIVATE
+                "${ANDROID_NDK}/sources/android/cpufeatures"
+                )
+        add_compile_definitions(
+                [[JUCE_ANDROID=1]]
+                [[JUCE_DONT_AUTO_OPEN_MIDI_DEVICES_ON_MOBILE=1]]
+                [[JUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]]
+                [[JUCE_PUSH_NOTIFICATIONS=1]]
+                [[JUCE_USE_ANDROID_OBOE=0]] # we don't need it at plugin side
+        )
+
+        target_link_libraries(${APP_NAME}
+                PRIVATE
+                ${log}
+                ${android}
+                ${glesv2}
+                ${egl}
+                ${cpufeatures_lib}
+                )
+endif (ANDROID)
+
+target_link_libraries(${APP_NAME}
+        PRIVATE
+        aap_audio_processors
+        )
+# <-- end Android specifics
+# <-- end AAP specifics
+
+target_link_libraries(SimpleReverb PUBLIC
     juce::juce_audio_basics
     juce::juce_audio_devices
     juce::juce_audio_formats
@@ -37,4 +90,4 @@ target_link_libraries(SimpleReverb PUBLIC
     juce::juce_gui_basics
     juce::juce_gui_extra)
 
-juce_generate_juce_header(SimpleReverb)
\ No newline at end of file
+juce_generate_juce_header(SimpleReverb)
diff --git a/External/JUCE b/External/JUCE
--- a/External/JUCE
+++ b/External/JUCE
@@ -1 +1 @@
-Subproject commit e6ec1819ec0a59a7cfe82d8ce72367f64e0a4bb0
+Subproject commit e6ec1819ec0a59a7cfe82d8ce72367f64e0a4bb0-dirty
